ansible/group_vars/.gitignore

```diff
@@ -1 +1 @@
-/all
+/*
```

Comment:
Is this trying to ignore everything in this directory in order to maintain an empty directory? If so, some tips to do that without ignoring the `.gitignore` file itself are at https://stackoverflow.com/a/932982/196405

Reply:
Looks like the `.gitignore` file is not ignored and still being tracked even if I don't explicitly whitelist `!.gitignore` as specified in the link. I have tested the below scenarios, let me know if I'm missing anything? Thanks

```
(env) [knarendran@kn-centos75-jbox fluo-muchos]$ git diff
diff --git a/ansible/group_vars/.gitignore b/ansible/group_vars/.gitignore
index 33662f5..4ef7868 100644
--- a/ansible/group_vars/.gitignore
+++ b/ansible/group_vars/.gitignore
@@ -1 +1,2 @@
-/*
+# Ignore everything in this directory
+*

(env) [knarendran@kn-centos75-jbox fluo-muchos]$ git status
# On branch karthick/azure_validations
# Changes not staged for commit:
#   (use "git add <file>..." to update what will be committed)
#   (use "git checkout -- <file>..." to discard changes in working directory)
#
#       modified:   ansible/group_vars/.gitignore
#
no changes added to commit (use "git add" and/or "git commit -a")
```

```
(env) [knarendran@kn-centos75-jbox fluo-muchos]$ git diff
diff --git a/ansible/group_vars/.gitignore b/ansible/group_vars/.gitignore
index 33662f5..5e7d273 100644
--- a/ansible/group_vars/.gitignore
+++ b/ansible/group_vars/.gitignore
@@ -1 +1,4 @@
-/*
+# Ignore everything in this directory
+*
+# Except this file
+!.gitignore

(env) [knarendran@kn-centos75-jbox fluo-muchos]$ git status
# On branch karthick/azure_validations
# Changes not staged for commit:
#   (use "git add <file>..." to update what will be committed)
#   (use "git checkout -- <file>..." to discard changes in working directory)
#
#       modified:   ansible/group_vars/.gitignore
#
no changes added to commit (use "git add" and/or "git commit -a")
```

Reply:
It looks like it works without excluding it because `*` doesn't match on dotfiles. However, that might be platform-dependent, since dotfiles are considered "hidden" on \*nix systems, but not necessarily on others. I'm not sure how git works across operating systems. If you wanted to ignore dotfiles, too, then you would probably have to add `.*` and `!.gitignore` also. I'm fine with whatever works, though.

Reply:
The objective is to commit the `.gitignore` file & ignore the rest of the files in that directory. Also, that is the only dotfile in the directory. In general, I think git doesn't ignore the `.gitignore` files. For instance, if we add `.*` or `.gitignore` git still doesn't ignore it and `git status` reports the file has been modified. Not sure if it's a git thing. Just to be safe, I'll go ahead and explicitly whitelist as mentioned in the above link.